dvc/repo/status.py

```diff
@@ -10,31 +10,32 @@
 logger = logging.getLogger(__name__)
 
 
-def _joint_status(stages):
+def _joint_status(pairs):
     status_info = {}
 
-    for stage in stages:
+    for stage, filter_info in pairs:
         if stage.frozen and not stage.is_repo_import:
             logger.warning(
                 "{} is frozen. Its dependencies are"
                 " not going to be shown in the status output.".format(stage)
             )
-
-        status_info.update(stage.status(check_updates=True))
+        if not filter_info:
+            status_info.update(stage.status(check_updates=True))
+        else:
+            for out in stage.filter_outs(filter_info):
+                status_info.update(out.status())
```

Comment:
Should we push this to `Stage::status()` or `Stage::_status_outs`?

Reply:
Also, we should keep `{stage_name: stage_status}` format, to make it in line with `--show-json`.

Reply:
> Thanks @karajan1001 for the PR. Looks good to me, although, have a few suggestions.
@skshetry , Thanks for your suggestions.

> Also, we should keep `{stage_name: stage_status}` format, to make it in line with `--show-json`.
Here, I just follow the format of `dvc status -c [outputs]`. Besides, if we show 

```
"alice_bob":  
        "changed outs": 
                "alice": "modified" 
```
We give misinformation that there is only one output "alice" in stage "alice_bob", here we need more emphasis on outputs than stages.

Reply:
@karajan1001, better to be consistent here. If we need better CLI output, we can handle that in `dvc/commands`. But, let's wait for others, let's see what they'll say. 

> It has a different format from the current Stage::status() and Stage::_status_outs now

Regarding  `status`, we could add `filter_info` to it so that you could just do
```suggestion
        status_info.update(stage.status(filter_outs=filter_info))
```

But this would make sense if we made result consistent.

Reply:
> @karajan1001, better to be consistent here. If we need better CLI output, we can handle that in `dvc/commands`. But, let's wait for others, let's see what they'll say.
> 
> > It has a different format from the current Stage::status() and Stage::_status_outs now
> 
> Regarding `status`, we could add `filter_info` to it so that you could just do
> 
> But this would make sense if we made result consistent.

@skshetry 
I found `filter_info` hard to use. Can we get rid of it? `collect_granular` returns a list of `outputs` and `stages`, and they share the same API `.status()` which returns the result. But seems that `filter_info` is at a finer granularity than outputs, as outputs may be a directory.

Reply:
I would agree with @skshetry here, 
```
"alice_bob":  
        "changed outs": 
                "alice": "modified" 
```
In that case we inform user that only one out has changed, but that does not mean that this particular stage has only this output. I would say that `status` primary function is to inform about changes, and there is no need to list all of its outputs, if they did not change.

Reply:
> I would agree with @skshetry here,
> 
> ```
> "alice_bob":  
>         "changed outs": 
>                 "alice": "modified" 
> ```
> 
> In that case we inform user that only one out has changed, but that does not mean that this particular stage has only this output. I would say that `status` primary function is to inform about changes, and there is no need to list all of its outputs, if they did not change.

@pared 
How about the `status -c`? Do they also need to follow this format, or just keep the current one?

Reply:
@karajan1001 This is a good question.
So in case of  our current test, if we commit modified `alice` we will get `Data and pipelines up to date` on `dvc status`
and we will get `new: alice` on `dvc status -c`. As this issue is about filtering the `status` I would keep output of `dvc status -c` as is and create issue discussing whether we should have consistent output format for both `dvc status` and `dvc status -c`.